### PR TITLE
Extensions: Try splitting CSS into separate files when building calypso

### PR DESCRIPTION
**Pull request by ryelle**
Inspired by talk of trimmimg down the main CSS file in calypso (see p4TIVU-8cI-p2), I've made this PR to enable extensions to build their CSS separately, rather than bundling into the main style.scss.

Specifically for the Store extension, we load a lot of CSS into the main style.css, when not many people are using Store (yet 😉). We can extract this to a separate file, and only load it when needed. This takes advantage of the existing work with section-specific CSS files and the fact that extensions define their own sections. 

There is now a new step in `build-css:sections`, a node script which checks if an extension calls for a css file ([woocommerce example](https://github.com/Automattic/wp-calypso/blob/try/extract-extension-css/client/extensions/woocommerce/package.json#L13)). This script knows that if there is a `css` entry, there is a `style.scss` file in the extension root that should be built into the `public/sections` folder. The script processed the scss and saves the CSS into `public/sections`. Then the rest of the `build-css:sections` steps take over, and autoprefixer & RTL are run on them.

This is a script, rather than a call directly to `node-sass`, because we need to filter out only the extensions that directly call for section-based CSS. It can't be enabled for all extensions by default because a) not all extensions have CSS, and b) each scss file will probably need to be updated to import the `shared/utils` file.

**TLDR;** This pulls out the woocommerce CSS into a new folder, and cuts almost 200K from `style.css`, and will be ready to use for any other extension to enable just by adding `css` to their section definition.

**BEFORE**

	1.5M public/style.css
	1.5M public/style-rtl.css

**AFTER**

	184K public/sections/woocommerce.css
	184K public/sections-rtl/woocommerce.rtl.css
	1.3M public/style.css
	1.3M public/style-rtl.css

**To test**

- Build calypso
- Check that there are new files in `public/sections` for woocommerce.css
- Load up calypso, keep the network tab open
- Select an atomic site with Store set up
- Click on Store in the sidebar, see the woocommerce.css file load
- Visit a store page directly: `/store/product/:site`
- See that the woocommerce.css file loads

### CODE REVIEW

**Comment by dmsnell** [comment]
this is good stuff @ryelle!

cc: @blowery since he has quite a bit of experience dealing with chunks and chunk hashes

**Comment by blowery** [comment]
Looks great to me. This doesn't directly impact webpack chunks afaict; hashes for css are generated outside of webpack. ( see https://github.com/Automattic/wp-calypso/blob/master/server/bundler/utils.js#L45 )

File: bin/process-extensions-scss.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+const fs = require( 'fs' );
+const path = require( 'path' );
+const sass = require( 'node-sass' );
+
+function getExtensionsWithScss() {
+	const extensionsDirectory = path.resolve( __dirname, '..', 'client', 'extensions' );
+	const extensionsNames = fs.readdirSync( extensionsDirectory ).filter( filename => {
+		// Return if this is a file (has extension)
+		if ( filename.indexOf( '.' ) !== -1 ) {
```
Comment (allendav) [inline bin/process-extensions-scss.js]: Would it be more robust to use isDirectory? https://stackoverflow.com/questions/15630770/node-js-check-if-path-is-file-or-directory

**Comment by stephanethomas** [comment]
Great job Kelly. Thanks for working on this!
